### PR TITLE
112: Add `/preview` route

### LINKED ISSUE #122: Create "preview" route for use in app like Feeder
Having a route dedicated to configuration preview in our apps will allow for more flexibility to adjust behavior and caching to fit expectations in those apps, without affecting and adding complexity to public use embeds.

### PR DESCRIPTION
**Pull request by rpeterman-gp**
Closes #122 

- preview allows for post message events to change config w/o reload
- preview sends post message with embed url, html, height, and styles
- move embed page content to embed component
- add helper functions to generate embed height and html style attibs
- improve embed height calcs and styles
- update caniuse db

## To Review

- [ ] Checkout Branch.
- [ ] Run `asdf install`.
- [ ] Run `yarn`.
- [ ] Run `yarn dev`.

> ...then...

- [ ] Have this dev server running when reviewing FEEDER_PR_LINK
- [ ] Ensure feeder form changes update the player
- [ ] Ensure height and styling set to feeder form cause the preview embed to properly size and scale based on config changes

### CODE REVIEW

File: components/Embed/Embed.tsx [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+                  })}
+                />
+              )}
+            </Player>
+          )}
+        </div>
+      </div>
+    </>
+  );
+};
```
Comment (cavis) [inline components/Embed/Embed.tsx:437]: Are there any changes in here, or largely copied whole from `pages/embed.tsx`?  (It's a big diff)

File: components/Player/PlayerShareMenu/PlayerShareMenu.tsx [hunk 2]
```diff
@@ -55,16 +57,20 @@ const PlayerShareMenu: React.FC<IPlayerShareMenuProps> = ({
 
           <CopyLinkButton />
 
-          <FileDownloadButton />
+          {!isPrivate && (
```
Comment (cavis) [inline components/Player/PlayerShareMenu/PlayerShareMenu.tsx:60]: I guess I would have preferred using the existing pattern - allowing the caller to override the download-button-url via a query parameter.  (Or passing an empty/blank value to indicate the button shouldn't be shown at all).  Would have been a bit more flexible.

But, many ways to do the same thing, and this gets the job done.

File: pages/preview.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      </Head>
+      <Embed config={newConfig} data={embedData} />
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<
+  IPreviewPageProps
+> = async ({ query, res }) => {
+  // 1. Convert query params into embed config.
```
Comment (cavis) [inline pages/preview.tsx:84]: So how much of this code is duplicative of `pages/embed.tsx`?  Just seems like a bit more lines of code than I would expect, for a page that's nearly identical.

**Review by cavis — APPROVED** [review]
👍 couple questions, but looks good to me

File: components/Embed/Embed.tsx (hunk 1 above)
Comment (rpeterman-gp) [reply components/Embed/Embed.tsx:437]: Yes. Basically just moved that markup and hooks into its own component that could be shared on the embed and preview pages.

File: components/Player/PlayerShareMenu/PlayerShareMenu.tsx (hunk 2 above)
Comment (rpeterman-gp) [reply components/Player/PlayerShareMenu/PlayerShareMenu.tsx:60]: That var is from the component props, but the value passed to the props coms from the embed config, which is parsed from the URL params. However, I didn't map the config object prop to a URL param key. We can easily add a param to the IEmbedConfig maps.

**Pull request merged**